scripts/db_operations.sh

```
@@ -0,0 +1,215 @@
+backup_db() {
+  log_message "info" "Creating database backup..."
+  check_db_connectivity || return 1
+  
+  timestamp=$(date +%Y%m%d_%H%M%S)
+  backup_file="/app/data/backups/db_backup_${timestamp}.sql"
+  
+  # Ensure backup directory exists
+  mkdir -p /app/data/backups
+  
+  # Perform the backup using pg_dump
+  PGPASSWORD=$POSTGRES_PASSWORD pg_dump -h $POSTGRES_HOST -p $POSTGRES_PORT -U $POSTGRES_USER -d $POSTGRES_DB > $backup_file
```

Shellcheck (0.10.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)